src/box/box.cc

```diff
@@ -782,28 +764,41 @@ box_check_election_timeout(void)
 	return d;
 }
 
-static void
-box_check_replication(void)
+static int
+box_check_uri_set(const char *option_name)
 {
-	int count = cfg_getarr_size("replication");
-	for (int i = 0; i < count; i++) {
-		const char *source = cfg_getarr_elem("replication", i);
-		if (box_check_uri(source, "replication") != 0)
-			diag_raise();
+	struct uri_set uri_set;
+	if (cfg_get_uri_set(option_name, &uri_set) != 0) {
```

Comment:
Is it correct that the entire `uri_set` export to Lua was done just to be able to get it here? Why couldn't you just store a plain Lua array of uri objects in Lua and save it into this 'set' in C in cfg.c file? We already have options working via Lua arrays. See `cfg_getarr_elem`.

Reply:
> Is it correct that the entire uri_set export to Lua was done just to be able to get it here?

No. `uri_set` is exposed to Lua (via ffi) in order to implement `uri.parse_many`. We don't use `uri.parse_many` in C code, we use `luaT_uri_set_create`, which creates a `uri_set` object from a Lua string or table.